internal/provider/resource/secret_resource.go

```diff
@@ -447,11 +447,12 @@ func (r *secretResource) Read(ctx context.Context, req resource.ReadRequest, res
 	} else if r.client.Config.IsMachineIdentityAuth {
 		// Get refreshed order value from HashiCups
 		response, err := r.client.GetSingleRawSecretByNameV3(infisical.GetSingleSecretByNameV3Request{
-			SecretName:  state.Name.ValueString(),
-			Type:        "shared",
-			WorkspaceId: state.WorkspaceId.ValueString(),
-			Environment: state.EnvSlug.ValueString(),
-			SecretPath:  state.FolderPath.ValueString(),
+			SecretName:             state.Name.ValueString(),
+			Type:                   "shared",
+			WorkspaceId:            state.WorkspaceId.ValueString(),
+			Environment:            state.EnvSlug.ValueString(),
+			SecretPath:             state.FolderPath.ValueString(),
+			ExpandSecretReferences: false,
```

Comment:
My (naïve) fear with defaulting to `true` here is that using a reference in a resource block might result in an inconsistent state after `apply` if the state switches to the resolved value:

```terraform
resource "infisical_secret" "blah" {
  name         = "SOME_SECRET"
  value        = "${prod.blah.value}" # this value would be different after apply
}
```